controller/src/main/java/io/pravega/controller/server/bucket/BucketManager.java

```diff
@@ -137,7 +147,7 @@ public void notify(BucketNotification notification) {
     }
 
     @VisibleForTesting
```

Comment:
A few lines up? Why does it do nothing in response to losing leadership?

Reply:
The leadership is acquired by way of creating an ephemeral znode. 
The only way it can lose leadership is if there is a zookeeper session expiration. 
In that case a controller wide shutdown is triggered. So there is nothing to handle there.